src/lib/storage/table.cpp

```diff
@@ -81,13 +97,62 @@ const std::string& Table::column_name(const ColumnID column_id) const { return _
 
 const std::string& Table::column_type(const ColumnID column_id) const { return _column_types.at(column_id); }
 
-std::shared_ptr<Chunk> Table::get_chunk(ChunkID chunk_id) { return _chunks.at(chunk_id); }
+std::shared_ptr<Chunk> Table::get_chunk(ChunkID chunk_id) {
+  auto guard = std::lock_guard<std::mutex>(_chunk_access_mutex);
+  return _chunks.at(chunk_id);
+}
 
-std::shared_ptr<const Chunk> Table::get_chunk(ChunkID chunk_id) const { return _chunks.at(chunk_id); }
+std::shared_ptr<const Chunk> Table::get_chunk(ChunkID chunk_id) const {
+  auto guard = std::lock_guard<std::mutex>(_chunk_access_mutex);
+  return _chunks.at(chunk_id);
+}
 
 void Table::compress_chunk(const ChunkID chunk_id) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  auto threads = std::vector<std::thread>{_column_types.size()};
```

Comment:
Employing column_count or column_names may be more intuitive as its not entirely clear from the member name that this is equivalent to the two.